xarray/plot/plot.py

```diff
@@ -542,29 +552,56 @@ def contourf(x, y, z, ax, **kwargs):
     return ax, primitive
 
 
-def _infer_interval_breaks(coord):
+def _infer_interval_breaks(coord, axis=0):
     """
     >>> _infer_interval_breaks(np.arange(5))
     array([-0.5,  0.5,  1.5,  2.5,  3.5,  4.5])
+    >>> _infer_interval_breaks([[0, 1], [0, 1]], axis=1)
+    array([[-0.5,  0.5,  1.5],
+           [-0.5,  0.5,  1.5]])
     """
     coord = np.asarray(coord)
-    deltas = 0.5 * (coord[1:] - coord[:-1])
-    first = coord[0] - deltas[0]
-    last = coord[-1] + deltas[-1]
-    return np.r_[[first], coord[:-1] + deltas, [last]]
+    if axis == 0:
+        deltas = 0.5 * (coord[1:, ...] - coord[:-1, ...])
```

Comment:
You could write an axis agnostic version of this with `np.diff`, `np.take` and `np.concatenate`, e.g.,
```
deltas = 0.5 * np.diff(coord, axis)
first = np.take(coord, [0], axis)
last = np.take(coord, [-1], axis)
trim_last = tuple(slice(None, -1) if n == axis else slice(None)
                  for n in range(coord.ndim))
return np.concatenate([first, coord[trim_last] + deltas, last], axis)
```

Maybe a slight improvement? (Up to you on whether to use it)